sequencer/src/block2.rs

```diff
@@ -0,0 +1,144 @@
+// Question: jellyfish uses `ark_std` everywhere instead of `std`.
+// Is there a chance of badness if we (as a downstream user) use `std`?
+use std::mem::size_of;
+
+#[allow(dead_code)] // TODO temporary
+pub struct BlockPayload {
+    payload: Vec<u8>,
+    // tx_ranges: Vec<Range<u32>>, // not convinced we need this
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Tx {
+    // namespace: u32, // TODO how many bytes for namespace id?
+    payload: Vec<u8>,
+}
+
+impl BlockPayload {
+    #[allow(dead_code)] // TODO temporary
+    fn build(txs: impl IntoIterator<Item = Tx>) -> Self {
+        // tx_table[i] is the end index (exclusive) of the ith tx
+        // so that the payload bytes of the ith tx is
+        // tx_bodies[tx_table[i-1]..tx_table[i]].
+        // edge case: tx_table[-1] is defined as 0.
+        //
+        // TODO clarification for future docs:
+        // perhaps the final entry of tx_table is redundant:
+        // it should always equal the length of the flattened tx bytes.
+        // But this might not always be the case.
+        // Example: If the flattened tx bytes are not a monolith at the end
+        // of the payload then the final tx_table entry is needed.
```

Comment:
We should know where the end of the payload or namespace is from the namespace table. I think it's fine to define the end of the last transaction as the end of that region. For an honest builder that should always be what you want anyways.

Reply:
Ok. That means for `n` txs we'll have only `n-1` entries in the tx table. That's good because it saves space but now there's a pesky edge case to deal with. I'll post a new issue for it.

Reply:
Yeah. Unclear if the extra complexity is worth it to save 4 bytes per namespace. We can circle back to this in the future.